### ISSUE TITLE
Fallback to second DNS Record is not even attempted in event of TCP Connection failure

### ISSUE THREAD

**Issue by desipepova** [Mobicents/sip-servlets#35]
Original issue URL from Google Code:

https://code.google.com/p/sipservlets/issues/detail?id=250&colspec=ID%20Status%20MSS%20Summary

**Comment by gkumar78** [comment]
Tested this issue on version mss-3.1.641-apache-tomcat-7.0.64 and it is working fine now. I configured 3 SRV records for SIP over TCP and all 3 were attempted in correct order of priority.

- If the tcp connection attempt was unsuccessful to a target, the next target was attempted right away.
- If tcp connection was successful but no final response was received within 32 seconds, the next target was attempted thereafter on expiry of 32 seconds.

Tested following 3 combinations and mobicents sip servlet handled them correctly.
1. No server listening on all 3 targets. The application received IO Exception immediately after container tried all the 3 targets and tcp connection to none was successful.
2. No server listening on first and third target, second target is listening but does not send any response. The application received 408 response after 32 seconds after container tried all the 3 targets. Tcp connection to first target was unsuccessful, and hence second target was attempted. After timeout of 32 seconds, third target was attempted which was also unsuccessful.
3. No server listening on first target, second target is listening but does not send any response, the third target is also listening and sends 200 OK response. The application received 200 OK response after 32 seconds. Tcp connection to first target was unsuccessful, and hence second target was attempted. After timeout of 32 seconds, third target was attempted which was successful.

This defect can be closed
